Tools/TestWebKitAPI/Tests/JavaScriptCore/glib/TestJSC.cpp

```diff
@@ -3963,6 +4001,23 @@ static void testJSCPromises()
         g_assert_true(jsc_value_is_number(value.get()));
         g_assert_cmpint(jsc_value_to_int32(value.get()), ==, -1);
     }
+
+    {
+        GMainLoop* m = g_main_loop_new(NULL, FALSE);
+
+        GRefPtr<JSCContext> context = adoptGRef(jsc_context_new());
+
+        JSCClass* foooClass = jsc_context_register_class(context.get(), "Fooo", nullptr, nullptr, reinterpret_cast<GDestroyNotify>(foooFree));
+        GRefPtr<JSCValue> foooConstructor = adoptGRef(jsc_class_add_constructor(foooClass, nullptr, G_CALLBACK(foooCreate), m, nullptr, G_TYPE_POINTER, 0, G_TYPE_NONE));
+        jsc_class_add_method(foooClass, "getPromise", G_CALLBACK(getPromise), nullptr, nullptr, JSC_TYPE_VALUE, 0, G_TYPE_NONE);
+        jsc_context_set_value(context.get(), jsc_class_get_name(foooClass), foooConstructor.get());
+
+        GRefPtr<JSCValue> result = adoptGRef(jsc_context_evaluate(context.get(), "result = -1; f = new Fooo(); f.getPromise().then(function (value) { result = 1; }, function (error) { result = 0; });", -1));
+        g_main_loop_run(m);
+        result = adoptGRef(jsc_context_get_value(context.get(), "result"));
+        g_assert_true(jsc_value_to_int32(result.get()) == 1);
```

Comment:
We need more test cases:
* Exception raised when creating the promise.
* Invoke then/catch methods from the API on the promise object.
* Resolve and rejection on promise creation and later
* Reject by raising an exception (both by setting the exception from the api and js)